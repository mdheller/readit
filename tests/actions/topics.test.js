import {
  UPVOTE_TOPIC,
  DOWNVOTE_TOPIC,
  NEW_TOPIC,
  upvoteTopic,
  downvoteTopic,
  createNewTopic,
} from '../../src/actions/topics';

describe('Constants', () => {
  it('matches snapshot values', () => {
    expect(UPVOTE_TOPIC).toMatchSnapshot();
    expect(DOWNVOTE_TOPIC).toMatchSnapshot();
    expect(NEW_TOPIC).toMatchSnapshot();
  });
});

describe('upvoteTopic', () => {
  it('returns the action in flux standard action format', () => {
    expect(upvoteTopic(1)).toMatchObject({
      type: UPVOTE_TOPIC,
      payload: 1,
    });
  });
  
  it('works with any type of id', () => {
    expect(upvoteTopic('1')).toMatchObject({
      type: UPVOTE_TOPIC,
      payload: '1',
    });

    expect(upvoteTopic(-1)).toMatchObject({
      type: UPVOTE_TOPIC,
      payload: -1,
    });

    expect(upvoteTopic()).toMatchObject({
      type: UPVOTE_TOPIC,
      payload: undefined,
    });
  });
});

describe('downvoteTopic', () => {
  it('returns the action in flux standard action format', () => {
    expect(downvoteTopic(1)).toMatchObject({
      type: DOWNVOTE_TOPIC,
      payload: 1,
    });
  });
  
  it('works with any type of id', () => {
    expect(downvoteTopic('1')).toMatchObject({
      type: DOWNVOTE_TOPIC,
      payload: '1',
    });

    expect(downvoteTopic(-1)).toMatchObject({
      type: DOWNVOTE_TOPIC,
      payload: -1,
    });

    expect(downvoteTopic()).toMatchObject({
      type: DOWNVOTE_TOPIC,
      payload: undefined,
    });
  });
});

describe('createNewTopic', () => {
  it('returns the action in flux standard action format', () => {
    expect(createNewTopic('test title')).toMatchObject({
      type: NEW_TOPIC,
      payload: 'test title',
    });
  });
  
  it('works with any type of title', () => {
    expect(createNewTopic('1')).toMatchObject({
      type: NEW_TOPIC,
      payload: '1',
    });

    expect(createNewTopic(-1)).toMatchObject({
      type: NEW_TOPIC,
      payload: -1,
    });

    expect(createNewTopic()).toMatchObject({
      type: NEW_TOPIC,
      payload: undefined,
    });
  });
});