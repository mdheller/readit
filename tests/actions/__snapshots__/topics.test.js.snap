// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Constants matches snapshot values 1`] = `"UPVOTE_TOPIC"`;

exports[`Constants matches snapshot values 2`] = `"DOWNVOTE_TOPIC"`;

exports[`Constants matches snapshot values 3`] = `"NEW_TOPIC"`;
